api/v1alpha1/gateway_types.go

```diff
@@ -326,15 +320,14 @@ type ListenerStatus struct {
 type ListenerConditionType string
 
 const (
-	// ConditionInvalidListener is a generic condition that is a catch all for
-	// unsupported configurations that do not match a more specific condition.
-	// Implementors should try to use a more specific condition instead of this
-	// one to give users and automation more information.
-	ConditionInvalidListener ListenerConditionType = "InvalidListener"
-	// ConditionListenerNotReady indicates the listener is not ready.
-	ConditionListenerNotReady ListenerConditionType = "ListenerNotReady"
-	// ConditionInvalidAddress indicates the Address is invalid.
-	ConditionInvalidAddress ListenerConditionType = "InvalidAddress"
+	// ConditionListenerValid is a generic condition that is a catch all for
+	// supported configurations. Implementors should try to use a more specific
+	// condition instead of this one to give users and automation more information.
+	ConditionListenerValid ListenerConditionType = "ListenerValid"
+	// ConditionListenerReady indicates the listener is ready to receive traffic.
+	ConditionListenerReady ListenerConditionType = "ListenerReady"
+	// ConditionValidAddress indicates the listener Address is valid.
+	ConditionValidAddress ListenerConditionType = "ValidAddress"
```

Comment:
Can we clarify that these will always overlap:

Valid implies Ready?
Or Valid && Ready == can serve traffic?